src/components/pill-selector-dropdown/Pill.js

```diff
@@ -3,19 +3,27 @@ import React from 'react';
 import classNames from 'classnames';
 
 type Props = {
+    isDisabled?: boolean,
     isSelected?: boolean,
     isValid?: boolean,
     onRemove: Function,
     text: string,
 };
 
-const Pill = ({ isSelected = false, isValid = true, onRemove, text }: Props) => (
-    <span className={classNames('pill', { 'is-selected': isSelected, 'is-invalid': !isValid })}>
-        <span className="pill-text">{text}</span>
-        <span aria-hidden="true" className="close-btn" onClick={onRemove}>
-            ✕
+const Pill = ({ isDisabled = false, isSelected = false, isValid = true, onRemove, text }: Props) => {
+    const styles = classNames('pill', {
+        'is-selected': isSelected && !isDisabled,
+        'is-invalid': !isValid,
+        'is-disabled': isDisabled,
+    });
+    return (
+        <span className={styles}>
+            <span className="pill-text">{text}</span>
+            <button aria-hidden="true" className="close-btn" disabled={isDisabled} onClick={onRemove} type="button">
```

Comment:
Just a heads-up: I think `PlainButton` handles disabled behavior a bit differently, via `event.preventDefault()`, though I'm not 100% sure why. Browsers also may style this button differently when it's disabled.

Reply:
Yep I tried to use `PlainButton` at first, but it required a lot more CSS overrides to get it to look visually the same as it currently does. I tested this with `disabled={true}` and it seemed to work as expected (prevented removal of the pill).